Crash in infoschema.FormatTiDBVersion

## Bug Report

We used the Fuzz engine to modify some Test(TestFormatVersion) data, and then the following crash appeared. We hope to get the help of the developer to confirm whether it is a real bug.

### 1. Minimal reproduce step (Required)

Simplified Test Code

```go
infoschema.FormatTiDBVersion("5.7.2Delta;5-TiDB-", true)
```

### 2. What did you expect to see? (Required)

Test Passed

### 3. What did you see instead (Required)

Crash Log

```go
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 125 [running]:
testing.tRunner.func1.2({0x373e4a0, 0xc0005bae70})
	/home/gogen/.local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/home/gogen/.local/go/src/testing/testing.go:1212 +0x218
panic({0x373e4a0, 0xc0005bae70})
	/home/gogen/.local/go/src/runtime/panic.go:1038 +0x215
github.com/pingcap/tidb/infoschema.FormatTiDBVersion({0x3939818, 0x12}, 0x0)
	/home/gogen/workspace/gowork/src/topproj/tidb/infoschema/tables.go:1615 +0x179
github.com/pingcap/tidb/infoschema_test.FuzzTestFormatVersion({0xc000436800, 0x22, 0x200})
	/home/gogen/workspace/gowork/src/topproj/tidb/infoschema/tables_test.go_TestFormatVersion_test.go:27 +0x9b
```

### 4. What is your TiDB version? (Required)

9c4f94dd70010806710153d27fd5b734f99c4cb1

--- a/infoschema/tables.go
+++ b/infoschema/tables.go
@@ -1729,8 +1729,8 @@ func FormatTiDBVersion(TiDBVersion string, isDefaultVersion bool) string {
 
 	// The user hasn't set the config 'ServerVersion'.
 	if isDefaultVersion {
-		nodeVersion = TiDBVersion[strings.LastIndex(TiDBVersion, "TiDB-")+len("TiDB-"):]
-		if nodeVersion[0] == 'v' {
+		nodeVersion = TiDBVersion[strings.Index(TiDBVersion, "TiDB-")+len("TiDB-"):]
+		if len(nodeVersion) > 0 && nodeVersion[0] == 'v' {
 			nodeVersion = nodeVersion[1:]
 		}
 		nodeVersions := strings.Split(nodeVersion, "-")

--- a/infoschema/tables_test.go
+++ b/infoschema/tables_test.go
@@ -676,8 +676,15 @@ func TestSelectHiddenColumn(t *testing.T) {
 
 func TestFormatVersion(t *testing.T) {
 	// Test for defaultVersions.
-	defaultVersions := []string{"5.7.25-TiDB-None", "5.7.25-TiDB-8.0.18", "5.7.25-TiDB-8.0.18-beta.1", "5.7.25-TiDB-v4.0.0-beta-446-g5268094af"}
-	defaultRes := []string{"None", "8.0.18", "8.0.18-beta.1", "4.0.0-beta"}
+	defaultVersions := []string{
+		"5.7.25-TiDB-None",
+		"5.7.25-TiDB-8.0.18",
+		"5.7.25-TiDB-8.0.18-beta.1",
+		"5.7.25-TiDB-v4.0.0-beta-446-g5268094af",
+		// “5.7.2Delta;5-TiDB-” is not a legal default TiDB Version, but shouldn't panic.
+		"5.7.2Delta;5-TiDB-",
+		"5.7.25-TiDB-v4.0.0-TiDB-446"}
+	defaultRes := []string{"None", "8.0.18", "8.0.18-beta.1", "4.0.0-beta", "", "4.0.0-TiDB"}
 	for i, v := range defaultVersions {
 		version := infoschema.FormatTiDBVersion(v, true)
 		require.Equal(t, defaultRes[i], version)

--- a/infoschema/tables_test.go
+++ b/infoschema/tables_test.go
@@ -681,8 +681,7 @@ func TestFormatVersion(t *testing.T) {
 		"5.7.25-TiDB-8.0.18",
 		"5.7.25-TiDB-8.0.18-beta.1",
 		"5.7.25-TiDB-v4.0.0-beta-446-g5268094af",
-		// “5.7.2Delta;5-TiDB-” is not a legal default TiDB Version, but shouldn't panic.
-		"5.7.2Delta;5-TiDB-",
+		"5.7.25-TiDB-",
 		"5.7.25-TiDB-v4.0.0-TiDB-446"}
 	defaultRes := []string{"None", "8.0.18", "8.0.18-beta.1", "4.0.0-beta", "", "4.0.0-TiDB"}
 	for i, v := range defaultVersions {

